applications/dashboard/models/class.logmodel.php

```diff
@@ -67,48 +67,42 @@ public static function endTransaction() {
     public function formatContent($Log) {
         $Data = $Log['Data'];
 
-        // TODO: Check for a custom log type handler.
-
-        switch ($Log['RecordType']) {
-            case 'Activity':
-                $Result = $this->formatKey('Story', $Data);
-                break;
-            case 'Discussion':
-                $Result =
-                    '<b>'.$this->formatKey('Name', $Data).'</b><br />'.
-                    $this->formatKey('Body', $Data);
-                break;
-            case 'ActivityComment':
-            case 'Comment':
-                $Result = $this->formatKey('Body', $Data);
-                break;
-            case 'Configuration':
-                $Result = $this->formatConfiguration($Data);
-                break;
-            case 'Registration':
-            case 'User':
-                $Result = $this->formatRecord(['Email', 'Name'], $Data);
-                if ($DiscoveryText = val('DiscoveryText', $Data)) {
-                    $Result .= '<br /><b>'.t('Why do you want to join?').'</b><br />'.Gdn_Format::display($DiscoveryText);
-                }
-                if (val('Banned', $Data)) {
-                    $Result .= "<br />".t('Banned');
-                }
-                break;
-            case 'Group':
-                $Result =
-                    '<b>'.$this->formatKey('Name', $Data).'</b><br />'.
-                    $this->formatKey('Description', $Data);
-                break;
-            case 'Event':
-                $Result =
-                    '<b>'.$this->formatKey('Name', $Data).'</b><br />'.
-                    $this->formatKey('Body', $Data).'<br />'.
-                    t('Location').': '.$this->formatKey('Location', $Data);
-                break;
-            default:
-                $Result = '';
+        $Result = '';
+        Gdn::pluginManager()->fireAs('LogModel')->fireEvent('FormatContent', [
```

Comment:
See later in this file for how to properly use event firing in this class:

                $L = self::instance();
                $L->EventArguments['Log'] =& $LogRow;
                $L->fireEvent('BeforeInsert');